code/game/objects/items/cards_ids.dm

```diff
@@ -901,6 +908,21 @@
 
 	return ..()
 
+
+/obj/item/card/id/advanced/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/our_crayon = W
+		if(tgui_alert(usr, "Recolor Department or Subdepartment?", "Recoloring ID...", list("Department", "Subdepartment")) == "Department")
+			if(do_after(user, 2 SECONDS)) // Doesn't technically require a spraycan's cap to be off but shhh
+				department_color_override = our_crayon.paint_color
+				balloon_alert(user, "recolored")
+			return
```

Comment:
This won't call update_icon(). That is not what I meant by early-return.

I meant:

```dm
if (!do_after(user, 2 SECONDS))
    return

// do stuff here

Reply:
I appear to be the stupid, my bad!
Addressing.

Reply:
You're not stupid